### PR TITLE
SVG editor

### PR DESCRIPTION
**Pull request by NeilFraser**

### CODE REVIEW

File: static/code/svg.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
+<html>
+  <head>
+    <title>Minimal demo of SvgCanvas</title>
```
Comment (cpcallen) [inline static/code/svg.html]: The initial check-in of SVG-edit should probably have *only* included the third-party code.

File: static/code/diff.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Mapping from diff operation to relevant tag.
+ */
+diffEditor.tagMap = {};
+diffEditor.tagMap[DIFF_INSERT] = 'ins';
+diffEditor.tagMap[DIFF_DELETE] = 'del';
+diffEditor.tagMap[DIFF_EQUAL] = 'span';
+
+window.addEventListener('load', diffEditor.init);
```
[the remark this answers is not in the document]
Comment (NeilFraser) [reply static/code/diff.js]: We are indeed in an iframe.

File: static/code/diff.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  var diff = this.dmp.diff_main(oldString, newString);
+  this.dmp.diff_cleanupSemantic(diff);
+  diffEditor.render(diff, document.body);
+};
+
+/**
+ * Return the current source.
+ * @return {string} Source.
+ */
+diffEditor.getString = function() {
```
[the remark this answers is not in the document]
Comment (NeilFraser) [reply static/code/diff.js]: I'd like the individual diffs to be revertable, thus making this a read/write editor.

File: static/code/diff.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<html>
+  <head>
+    <title>Code City Diff Editor</title>
+    <script src="mobwrite/dmp.js"></script>
```
[the remark this answers is not in the document]
Comment (NeilFraser) [reply static/code/diff.html]: Mobwrite is incoming...

File: static/code/diff.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (cpcallen) [reply static/code/diff.js]: Oh, O.K.  That would be useful.

**Comment by NeilFraser** [comment]
Closing in favour of a new branch.

**Pull request closed without merging**